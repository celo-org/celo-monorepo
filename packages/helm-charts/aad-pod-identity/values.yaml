namespace: default

nmi:
  forceNamespaced: true
  prometheusPort: 9090
  image:
    repo: mcr.microsoft.com/k8s/aad-pod-identity/nmi
    tag: 1.6.2
  resources:
    limits:
      cpu: 200m
      memory: 512Mi
    requests:
      cpu: 100m
      memory: 256Mi
mic:
  syncRetryDuration: 1m
  forceNamespaced: true
  prometheusPort: 9090
  image:
    repo: mcr.microsoft.com/k8s/aad-pod-identity/mic
    tag: 1.6.2
  resources:
    limits:
      cpu: 200m
      memory: 1024Mi
    requests:
      cpu: 100m
      memory: 256Mi
