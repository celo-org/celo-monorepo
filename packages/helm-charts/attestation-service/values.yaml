imagePullPolicy: IfNotPresent

# Node labels for pod assignment
# ref: https://kubernetes.io/docs/user-guide/node-selection/
nodeSelector: {}

