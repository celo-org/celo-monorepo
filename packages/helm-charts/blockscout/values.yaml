imagePullPolicy: IfNotPresent
promtosd:
  image:
    repository: gcr.io/google-containers/prometheus-to-sd
    tag: v0.3.2
  port: 6060
blockscout:
  image:
    repository: gcr.io/celo-testnet/blockscout
    tag: v2.0.4-beta-celo
  db:
    # ip: must be provided at runtime # IP address of the postgres DB
    # connection_name: must be provided at runtime # name of the cloud sql connection
    # username: blockscout
    # password: password
    name: blockscout
  jsonrpc_http_url: http://tx-nodes:8545
  jsonrpc_ws_url: ws://tx-nodes:8546
