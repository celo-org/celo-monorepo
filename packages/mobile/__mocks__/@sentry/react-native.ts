const Sentry = {
  config: () => ({ install: jest.fn() }),
  setTagsContext: jest.fn(),
  captureException: jest.fn(),
}

module.exports = Sentry
