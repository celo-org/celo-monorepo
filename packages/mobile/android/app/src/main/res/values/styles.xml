<resources>
  <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
    <item name="android:windowLightNavigationBar">true</item>
    <item name="android:navigationBarColor">@android:color/white</item>
    <item name="android:textColor">#000000</item>
  </style>
</resources>
