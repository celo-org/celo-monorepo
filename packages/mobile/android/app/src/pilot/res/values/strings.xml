<resources>
  <string name="app_name">Monedero Celo</string>
</resources>
