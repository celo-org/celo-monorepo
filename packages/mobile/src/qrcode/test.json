{
    "address": "0xf7f551752A78Ce650385B58364225e5ec18D96cB",
    "displayName": "Super 8",
    "currencyCode": "PHP",
    "amount": "500",
    "comment": "92a53156-c0f2-11ea-b3de-0242ac13000"
}