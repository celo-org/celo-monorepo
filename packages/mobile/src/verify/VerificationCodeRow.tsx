import { stripHexLeader } from '@celo/utils/src/address'
import { extractAttestationCodeFromMessage } from '@celo/walletkit'
import * as React from 'react'
import CodeRow, { CodeRowStatus } from 'src/components/CodeRow'
import { ATTESTATION_CODE_PLACEHOLDER } from 'src/identity/reducer'
import { AttestationCode } from 'src/identity/verification'
import Logger from 'src/utils/Logger'

export const CODE_INPUT_PLACEHOLDER = '<#> celo://wallet/v/m9o...'

interface OwnProps {
  index: number // index of code in attestationCodes array
  inputValue: string // the raw code inputed by the user
  isCodeSubmitting: boolean // is the inputted code being processed
  onInputChange: (value: string) => void
  attestationCodes: AttestationCode[] // The codes in the redux store
  numCompleteAttestations: number // has the code been accepted and completed
}

function VerificationCodeRow({
  index,
  inputValue,
  onInputChange,
  isCodeSubmitting,
  attestationCodes,
  numCompleteAttestations,
}: OwnProps) {
  let codeStatus: CodeRowStatus = CodeRowStatus.DISABLED
  if (numCompleteAttestations > index) {
    codeStatus = CodeRowStatus.ACCEPTED
    inputValue = getRecodedAttestationValue(attestationCodes[index])
  } else if (attestationCodes.length > index) {
    codeStatus = CodeRowStatus.RECEIVED
    inputValue = getRecodedAttestationValue(attestationCodes[index])
  } else if (isCodeSubmitting) {
    codeStatus = CodeRowStatus.PROCESSING
  } else if (attestationCodes.length === index) {
    codeStatus = CodeRowStatus.INPUTTING
  }

  return (
    <CodeRow
      status={codeStatus}
      inputValue={inputValue}
      inputPlaceholder={CODE_INPUT_PLACEHOLDER}
      onInputChange={onInputChange}
      shouldShowClipboard={shouldShowClipboard(attestationCodes)}
    />
  )
}

function getRecodedAttestationValue(attestationCode: AttestationCode) {
  try {
    if (!attestationCode.code || attestationCode.code === ATTESTATION_CODE_PLACEHOLDER) {
      return ''
    }
    return Buffer.from(stripHexLeader(attestationCode.code), 'hex').toString('base64')
  } catch (error) {
    Logger.warn('VerificationCodeRow', 'Could not recode verification code to base64')
    return ''
  }
}

function shouldShowClipboard(attestationCodes: AttestationCode[]) {
  return (value: string) => {
    const extractedCode = extractAttestationCodeFromMessage(value)
    return !!extractedCode && !attestationCodes.find((c) => c.code === extractedCode)
  }
}

export default VerificationCodeRow
