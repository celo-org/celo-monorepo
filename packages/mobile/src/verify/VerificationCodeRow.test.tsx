import * as React from 'react'
import 'react-native'
import { render } from 'react-native-testing-library'
import VerificationCodeRow from 'src/verify/VerificationCodeRow'
import { mockAttestationMessage } from 'test/values'

describe('VerificationCodeRow', () => {
  it('renders correctly when input enabled', () => {
    const { toJSON } = render(
      <VerificationCodeRow
        index={0}
        attestationCodes={[]}
        isInputEnabled={true}
        inputValue={'test'}
        onInputChange={jest.fn()}
        isCodeSubmitting={false}
        isCodeAccepted={false}
      />
    )
    expect(toJSON()).toMatchSnapshot()
  })
  it('renders correctly when input disabled', () => {
    const { toJSON } = render(
      <VerificationCodeRow
        index={0}
        attestationCodes={[]}
        isInputEnabled={false}
        inputValue={'test'}
        onInputChange={jest.fn()}
        isCodeSubmitting={false}
        isCodeAccepted={false}
      />
    )
    expect(toJSON()).toMatchSnapshot()
  })
  it('renders correctly when submitting', () => {
    const { toJSON } = render(
      <VerificationCodeRow
        index={0}
        attestationCodes={[]}
        isInputEnabled={true}
        inputValue={'test'}
        onInputChange={jest.fn()}
        isCodeSubmitting={true}
        isCodeAccepted={false}
      />
    )
    expect(toJSON()).toMatchSnapshot()
  })
  it('renders correctly when code received', () => {
    const { toJSON } = render(
      <VerificationCodeRow
        index={0}
        attestationCodes={[mockAttestationMessage]}
        isInputEnabled={true}
        inputValue={'test'}
        onInputChange={jest.fn()}
        isCodeSubmitting={false}
        isCodeAccepted={false}
      />
    )
    expect(toJSON()).toMatchSnapshot()
  })
  it('renders correctly when code accepted', () => {
    const { toJSON } = render(
      <VerificationCodeRow
        index={0}
        attestationCodes={[mockAttestationMessage]}
        isInputEnabled={true}
        inputValue={'test'}
        onInputChange={jest.fn()}
        isCodeSubmitting={false}
        isCodeAccepted={true}
      />
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
