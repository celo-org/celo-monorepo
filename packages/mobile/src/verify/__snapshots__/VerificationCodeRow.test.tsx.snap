// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerificationCodeRow renders correctly when code accepted 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#EEEEEE",
      "borderRadius": 3,
      "height": 50,
      "justifyContent": "center",
      "marginVertical": 5,
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "color": "#81868B",
        "fontFamily": "Hind-Regular",
        "fontSize": 15,
        "lineHeight": 24,
      }
    }
  >
    q4BJuVrALpiaroth/dwQ/ps6w8auvNPmi+SVVwstPaFaq8aRq4jeaWSPmI+rZTrJQ/Z0BOUyz9EBNif1Y2XzZQE=
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#EEEEEE",
        "padding": 10,
        "position": "absolute",
        "right": 3,
        "top": 3,
      }
    }
  >
    <svg
      fill="none"
      height={20}
      style={Object {}}
      viewBox="0 0 18 18"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="#45CD86"
        height="2"
        rx="1"
        style={Object {}}
        transform="rotate(-135 6.89185 17.0205)"
        width="8.33324"
        x="6.89185"
        y="17.0205"
      />
      <rect
        fill="#45CD86"
        height="2"
        rx="1"
        style={Object {}}
        transform="rotate(-45 5.47778 15.6064)"
        width="14.9998"
        x="5.47778"
        y="15.6064"
      />
    </svg>
  </View>
</View>
`;

exports[`VerificationCodeRow renders correctly when code received 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#EEEEEE",
      "borderRadius": 3,
      "height": 50,
      "justifyContent": "center",
      "marginVertical": 5,
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "color": "#81868B",
        "fontFamily": "Hind-Regular",
        "fontSize": 15,
        "lineHeight": 24,
      }
    }
  >
    q4BJuVrALpiaroth/dwQ/ps6w8auvNPmi+SVVwstPaFaq8aRq4jeaWSPmI+rZTrJQ/Z0BOUyz9EBNif1Y2XzZQE=
  </Text>
</View>
`;

exports[`VerificationCodeRow renders correctly when input disabled 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F0F0F0",
      "borderColor": "#D1D5D8",
      "borderRadius": 3,
      "borderWidth": 1,
      "height": 50,
      "justifyContent": "center",
      "marginVertical": 5,
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#81868B",
        "fontFamily": "Hind-Regular",
        "fontSize": 15,
        "lineHeight": 24,
      }
    }
  >
    &lt;#&gt; m9oASm/3g7aZ...
  </Text>
</View>
`;

exports[`VerificationCodeRow renders correctly when input enabled 1`] = `
<View
  style={
    Object {
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#FFF",
            "borderColor": "#D1D5D8",
            "borderRadius": 3,
            "borderWidth": 1,
            "flex": 0,
            "height": 50,
            "marginVertical": 5,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        onFocus={[Function]}
        placeholder="<#> m9oASm/3g7aZ..."
        rejectResponderTermination={true}
        shouldShowClipboard={[Function]}
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#D1D5D8",
            "borderRadius": 3,
            "flex": 1,
            "fontFamily": "Hind-Regular",
            "padding": 8,
          }
        }
        underlineColorAndroid="transparent"
        value="test"
      />
    </View>
  </View>
</View>
`;

exports[`VerificationCodeRow renders correctly when submitting 1`] = `
<View
  style={
    Object {
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#FFF",
            "borderColor": "#D1D5D8",
            "borderRadius": 3,
            "borderWidth": 1,
            "flex": 0,
            "height": 50,
            "marginVertical": 5,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        onBlur={[Function]}
        onChangeText={[MockFunction]}
        onFocus={[Function]}
        placeholder="<#> m9oASm/3g7aZ..."
        rejectResponderTermination={true}
        shouldShowClipboard={[Function]}
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#D1D5D8",
            "borderRadius": 3,
            "flex": 1,
            "fontFamily": "Hind-Regular",
            "padding": 8,
          }
        }
        underlineColorAndroid="transparent"
        value="test"
      />
    </View>
  </View>
  <ActivityIndicator
    animating={true}
    color="#42D689"
    hidesWhenStopped={true}
    size="small"
    style={
      Object {
        "backgroundColor": "#FFF",
        "padding": 10,
        "position": "absolute",
        "right": 3,
        "top": 9,
      }
    }
  />
</View>
`;
